blueflood-core/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/RollupHandler.java

```diff
@@ -206,4 +268,28 @@ protected MetricData getHistogramsByGranularity(String tenantId,
 
         return data;
     }
+
+    private class UnitsGetter implements Callable {
+        private String tenantId;
+        private String metricName;
+
+        public UnitsGetter(String tenantId, String metricName) {
+            this.tenantId = tenantId;
+            this.metricName = metricName;
+        }
+
+        public String call() {
+            String unit = null;
+            try {
+                List<SearchResult> results = discoveryHandler.search(tenantId, metricName);
+                for (SearchResult res : results) {
```

Comment:
@tilogaat instead of this iteration, can we just do `results.get(0).getUnit()`? Also, it will be good to assert in some way that the results' size is always 1, may be have a block of code which tests that?